flocker/node/agents/blockdevice.py

```diff
@@ -801,6 +943,16 @@ class CreateBlockDeviceDataset(PRecord):
     dataset = field(mandatory=True, type=Dataset)
     mountpoint = field(mandatory=True, type=FilePath)
 
+    @classmethod
+    def from_state_and_config(cls, discovered_dataset, desired_dataset):
+        return cls(
+            dataset=Dataset(
```

Comment:
The implementation of `run` relies on the metadata to figure out if there's a profile, so you need that as well.